scripts/pr-infra-compose.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+# This script enables the deployment of a dedicated app instance using the
+# following inputs:
+#   * PR Number: Specify the pull request number for the instance you wish to
+#       run.
+#   * Infrastructure Backup: Provide a backup of the infrastructure to load
+#       into the app.
+#
+# Key Benefits:
+#   * Isolated Environment: The new instance operates independently of the
+#       development environment, allowing both to run in parallel without
+#       interference.
+#   * Faster Reviews: The script pulls Docker images generated by CI,
+#       eliminating the need to rebuild everything from scratch, which
+#       accelerates the review process.
+
+
+set -e
+
+# Usage
+
+if [ "$#" -ne 3 ] ||
+    { [ "$3" != "up" ] && [ "$3" != "down" ]; }; then
+    echo "Usage: $0 pr-number osrd.backup (up | down)" >&2
+    exit 1
+fi
+
+# Change ports in needed files
+export PR_NB="pr-$1"
+
+if [ "$3" = "up" ]; then
```

Comment:
I added the first suggestions.

I'm not sure that we should combine the script with `osrd-compose.sh` as they don't have the same purpose, it would make the `osrd-compose.sh` a lot more complex for not so much added value (imo)
Or maybe I didn't understand your last suggestion ?

Reply:
Nevermind, I messed up with git, the change isn't on the upstream yet

Reply:
It is now ;)

Reply:
> Then, maybe mutualize with osrd-compose.sh and just allow for different project-name and file?

I feel osrd-compose was more a hack to have the host mode accessible more easily (its name is kinda bad, I should not have named this script osrd-compose in the first place).

Reply:
Thanks for the rename, the use should be less confusing that way 🙏 

@ElysaSrc : renaming (and changing doc if necessary) osrd-compose.sh to `host-compose.sh` may be more consistent indeed.

Finally, as discussed a bit yesterday, my idea was to "intercept" `up-and-load-backup` and `down-and-clean`, and in all other cases fallback to just `docker compose "$@"` (maybe not so easy to document and create a `--help`).
So enrich the docker compose options in a script, and then from this script offer host-compose and pr-infra-compose that would just provide appropriate `-p project` and `-f file`.
That way we would have one way of doing all and also allow for `up-and-load-backup`|`down-and-clean` on a "classic" or on a "host" setup.
This was my idea, but it's only a nice-to-have IMO, so feel free to let it aside.